server/service/Cargo.toml

```diff
@@ -24,10 +25,12 @@ tera = "1"
 tokio = "1.17.0"
 headless_chrome = "0.9"
 failure = "0.1.8"
+actix-web = { version= "4.0.1" } 
```

Comment:
Slightly unclean to have actix-web in the service.

As far as I have seen its used for tests and in

```rust 
pub struct Synchroniser {
    settings: SyncSettings,
    // HERE: Data<> can be replaced with Arc<ServiceProvider>
    service_provider: Data<ServiceProvider>,
    pub(crate) central_data: CentralDataSynchroniser,
    pub(crate) remote_data: RemoteDataSynchroniser,
}
```
The use of Data can be replaced with an `Arc` using `Data::into_inner`:
```rust
impl<T: ?Sized> Data<T> {
    /// Returns reference to inner `T`.
    pub fn get_ref(&self) -> &T {
        self.0.as_ref()
    }

    /// Unwraps to the internal `Arc<T>`
    pub fn into_inner(self) -> Arc<T> {
        self.0
    }
}
```

This means we can move the the actix-web dep to [dev-dependencies]

Reply:
sure, was thinking it might be cleaner to use same type for service provider boxing everywhere, but I see your point: 2d529ac0960d578ebe3da8e4dab62ccdf36eb6e1